src/date-range-picker/__tests__/utils.test.ts

```diff
@@ -0,0 +1,295 @@
+    });
+
+    test('calls setTimeOffset for default case', () => {
+      const input = {
+        type: 'absolute',
+        startDate: '2023-06-15T00:00:00Z',
+        endDate: '2023-07-20T23:59:59Z',
+      };
+      const timeOffset = { startDate: 3600000, endDate: 7200000 };
+
+      formatValue(input as DateRangePickerProps.Value, { ...defaultOptions, timeOffset });
+
+      expect(SetTimeOffsetSpy).toHaveBeenCalledWith(input, timeOffset);
+    });
+
+    it('should return null when value is null', () => {
+      const result = formatValue(null, { timeOffset: {}, dateOnly: false });
+      expect(result).toBeNull();
+    });
+
+    it('should return the value when type is relative', () => {
+      const value: DateRangePickerProps.Value = { type: 'relative', amount: 7, unit: 'day' };
+      const result = formatValue(value, { timeOffset: {}, dateOnly: false });
+      expect(result).toBe(value);
+    });
+
+    it('should return date-only format when dateOnly is true', () => {
+      const value: DateRangePickerProps.Value = {
+        type: 'absolute',
+        startDate: '2023-01-01T12:00:00',
+        endDate: '2023-01-07T12:00:00',
+      };
+      const result = formatValue(value, { timeOffset: {}, dateOnly: true });
+      expect(result).toEqual({ type: 'absolute', startDate: '2023-01-01', endDate: '2023-01-07' });
+    });
+
+    it('should call setTimeOffset when dateOnly is false', () => {
+      const value: DateRangePickerProps.Value = {
+        type: 'absolute',
+        startDate: '2023-01-01T12:00:00',
+        endDate: '2023-01-07T12:00:00',
+      };
+      const timeOffset = { startDate: 0, endDate: 0 };
+      formatValue(value, { timeOffset, dateOnly: false });
+      expect(SetTimeOffsetSpy).toHaveBeenCalledWith(value, timeOffset);
+    });
+  });
+
+  describe('formatInitialValue', () => {
+    const normalizedTimeOffset = { startDate: 0, endDate: 0 };
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should return shifted time offset when value is null', () => {
+      const shiftedValue: DateRangePickerProps.AbsoluteValue = {
+        type: 'absolute',
+        startDate: '2023-01-01',
+        endDate: '2023-01-31',
+      };
+      ShiftTimeOffsetSpy.mockReturnValue(shiftedValue);
+
+      const result = formatInitialValue(null, false, normalizedTimeOffset);
+
+      expect(ShiftTimeOffsetSpy).toHaveBeenCalledWith(null, normalizedTimeOffset);
+      expect(result).toBe(shiftedValue);
+    });
+
+    it('should return shifted time offset for relative value', () => {
+      const relativeValue: DateRangePickerProps.RelativeValue = { type: 'relative', amount: 7, unit: 'day' };
+      const shiftedValue: DateRangePickerProps.AbsoluteValue = {
+        type: 'absolute',
+        startDate: '2023-01-01',
+        endDate: '2023-01-07',
+      };
+      ShiftTimeOffsetSpy.mockReturnValue(shiftedValue);
```

Comment:
My understanding is that `shiftTimeOffset` will _not_ shift a relative value ([code](https://github.com/cloudscape-design/components/blob/edb6713937169bbd4c25efb540fa6361f849bb2d/src/date-range-picker/time-offset.ts#L36-L38)) - this mocked value seems to be a value that will never be returned in the actual component, it will be the `relativeValue` itself. Can we update this test?

Reply:
Because the shiftTimeOffset was mocked it never went into that branch. I added more tests to the shiftTimeOffset function to make assertions when there is a relative  type and here I have the mock returning the correct value